### PR TITLE
Improve tsq/tree-sitter-query language support

### PR DESCRIPTION
**Pull request by uncenter**
WIP adding support for tree-sitter query files (scheme-like), using https://github.com/tree-sitter-grammars/tree-sitter-query for the grammar and https://github.com/ribru17/ts_query_ls as the LSP.

## Todo

- [x] LSP
- [x] Reverse query order for precedence order difference

### CODE REVIEW

**Review by the-mikedavis — COMMENTED** [review]
You can update the existing entry for this language instead: https://github.com/helix-editor/helix/blob/b1bdbc6789a66149d6c7710371d25e2351891305/languages.toml#L1439-L1449

That language is set to recognize `.tsq` files as a dummy value. (We didn't want the tree-sitter query language to be used over Scheme generally.) The file-types you have proposed here instead looks better though, there might still be some overlap with regular Scheme files but it seems unlikely.

**Comment by the-mikedavis** [comment]
You may need to set `grammar = "query"` in the `[[language]]` entry for `tsq`. The `[[grammar]]` entry's name is used for loading the parser so it needs to basically match the `name` attribute from the `grammar.js` https://github.com/tree-sitter-grammars/tree-sitter-query/blob/a6674e279b14958625d7a530cabe06119c7a1532/grammar.js#L16

File: runtime/queries/tsq/highlights.scm [hunk 1]
```diff
@@ -1,50 +1,78 @@
...
- (#match? @keyword.directive "^; +inherits *:"))
+((comment) @keyword.directive
+  (#match? @keyword.directive "^;+\s*format\-ignore\s*$"))
+
+((program
+  .
+  (comment)*
+  .
+  (comment) @keyword.directive)
+  (#match? @keyword.directive "^;+ *extends *$"))
```
Comment (the-mikedavis) [inline runtime/queries/tsq/highlights.scm:20]: I believe these are specific to neovim's queries so they can be removed

File: runtime/queries/tsq/highlights.scm [hunk 2]
```diff
@@ -1,50 +1,78 @@
...
+((program
+  .
+  (comment)*
+  .
+  (comment) @keyword.import)
+  (#match? @keyword.import "^;+ *inherits *:"))
+
+((parameters
+  (identifier) @number)
+  (#match? @number "^[-+]?[0-9]+(.[0-9]+)?$"))
```
Comment (the-mikedavis) [inline runtime/queries/tsq/highlights.scm:31]:
```suggestion
  (identifier) @constant.numeric)
  (#match? @constant.numeric "^[-+]?[0-9]+(.[0-9]+)?$"))
```

File: runtime/queries/tsq/highlights.scm [hunk 3]
```diff
@@ -1,50 +1,78 @@
...
 
 (comment) @comment
 
-(field_name) @variable.other.member
+(negated_field
+  "!" @operator
+  (identifier) @property)
+
+(field_definition
+  name: (identifier) @property)
```
Comment (the-mikedavis) [inline runtime/queries/tsq/highlights.scm:62]:
```suggestion
  (identifier) @variable.other.member)

(field_definition
  name: (identifier) @variable.other.member)
```

File: runtime/queries/tsq/injections.scm [hunk 4]
```diff
@@ -1,8 +1,10 @@
...
 ((predicate
-   (predicate_name) @_predicate
-   (string) @injection.content)
- (#eq? @_predicate "#match?")
- (#set! injection.language "regex"))
+  name: (identifier) @_name
+  parameters:
+    (parameters
+      (string) @injection.content))
+  (#any-of? @_name "match" "not-match" "vim-match" "not-vim-match")
```
Comment (the-mikedavis) [inline runtime/queries/tsq/injections.scm:9]:
```suggestion
  (#any-of? @_name "match" "not-match")
```

the vim ones are specific to neovim, I believe they use a different regex flavor

Comment (uncenter) [reply runtime/queries/tsq/injections.scm:9]: Yes, but we can still highlight these function names the same.

**Review by the-mikedavis — APPROVED** [review]
thanks!

**Pull request merged**